runtime-modules/content/src/benchmarks/benchmarking.rs

```diff
@@ -183,7 +190,75 @@ benchmarks! {
         );
     }
 
-    delete_channel {
+    channel_update_without_assets {
+
+        let a in 1 .. T::MaxNumberOfCollaboratorsPerChannel::get(); //max colaborators
+
+        let b in (T::StorageBucketsPerBagValueConstraint::get().min as u32) ..
+         (T::StorageBucketsPerBagValueConstraint::get().max() as u32);
+
+        let c in
+            (T::DistributionBucketsPerBagValueConstraint::get().min as u32) ..
+            (T::DistributionBucketsPerBagValueConstraint::get().max() as u32);
```

Comment:
For `channel_update_without_assets` and `channel_delete_with_assets` we need some modifications in the content pallet, so that the weights don't depend on storage/distribution buckets number in that case.

Basically you need to put everything related to data object uploading/removing in:
```rust
if params.assets_to_upload.is_some() || !params.assets_to_remove.is_empty() {
  // ...
}
```
Especially `upload_and_delete_data_objects`.

You can also make `channel_bag_witness` optional and only check it if `params.assets_to_upload.is_some() || !params.assets_to_remove.is_empty()`

Reply:
do we need this "if" in delete_channel? 
The `num_objects_to_delete` is used here

https://github.com/Joystream/joystream/blob/12cd1645806644280b15269c3afe51f82730f77c/runtime-modules/content/src/lib.rs#L730

and is validating only if the number of the objects in the bag, matches the number we want to delete

https://github.com/Joystream/joystream/blob/12cd1645806644280b15269c3afe51f82730f77c/runtime-modules/content/src/lib.rs#L3288-L3291

should we always remove the bag? even if there are no objects?
I need to write a test where the objects are 0 but the fixture is validating at the end, if the bag exists

Reply:
> should we always remove the bag? even if there are no objects?

Oh, right, I kept having `delete_video` on my mind, where there was no need to remove the bag.

Here we need to remove the entire channel bag, so we always need to update all storage/distribution buckets anyway. This means there's really no point in splitting `delete_channel` to "with_assets" and "without_assets" cases. Sorry for a bad suggestion.